Makefile.inc

```diff
@@ -1169,6 +1176,10 @@ endif # CONFIG_CPU_INTEL_FIRMWARE_INTERFACE_TABLE
 	$(CBFSTOOL) $@ layout
 	@printf "    CBFSPRINT  $(subst $(obj)/,,$(@))\n\n"
 	$(CBFSTOOL) $@ print -r $(subst $(spc),$(comma),$(all-regions))
+ifeq ($(CONFIG_ARCH_PPC64),y)
+	$(ECCTOOL) --inject $@ --output $@.ecc --p8
```

Comment:
Unfortunately, we need to add header with signing before applying ECC. We would have to use https://github.com/open-power/sb-signing-utils, or otherwise craft that header. IIUC it will be constant except for "HBB" hash and size, so maybe it would be easier to do this manually than add another utils.

Reply:
You mean something like here? https://github.com/open-power/pnor/blob/master/update_image.pl#L547

Reply:
Kind of, though the header is different in image obtained from the machine:
```
00205000: 1708 2011 0001 0000 7600 0000 0850 2000  .. .....v....P .
00205010: 00ff 0000 0000 0000 0000 0000 0000 0000  ................
00205020: 0000 0000 0000 0000 0000 0000 0000 0000  ................
00205030: 0000 0000 0000 0000 0000 0000 0000 0000  ................
```
This is already after ECC so discard every 9th byte.

Reply:
But shouldn't these headers be present for HBBL? not HBB?

Reply:
The dump I pasted is for HBB

Reply:
I can't find the header from `update_image.pl` anywhere in the image

Reply:
@krystian-hebel i have found how it is constructed. Added signing with submodule. Now the header is:
```
$ hexdump -C build/coreboot.rom.signed.ecc |head
00000000  17 08 20 11 00 01 00 00  76 00 00 00 08 10 00 00  |.. .....v.......|
00000010  00 93 00 00 00 00 00 00  00 00 00 00 00 00 00 00  |................|
00000020  00 00 bb 78 1e 08 78 96  a0 9e 30 72 db 74 06 8d  |...x..x...0r.t..|
00000030  e7 ca 8a 02 a0 6f 9c 55  43 8f 50 f4 de 29 8e 1e  |.....o.UC.P..)..|
00000040  63 37 9f 73 6c b7 c2 f2  7a 1e f2 77 b2 78 1f 97  |c7.sl...z..w.x..|
00000050  6c d3 bd 64 a5 78 3c de  71 aa 00 56 ec 6a 9b 56  |l..d.x<.q..V.j.V|
00000060  27 d4 c5 83 09 08 ff 53  cf b3 61 0a 00 b8 60 94  |'......S..a...`.|
00000070  41 76 47 37 40 22 51 2c  05 f8 60 f1 f0 b8 25 bb  |AvG7@"Q,..`...%.|
00000080  46 54 81 97 16 ed 94 10  fc 69 38 30 fc fe d2 04  |FT.......i80....|
00000090  26 9e 34 62 83 f3 a7 81  db 91 5c 7b c7 dd b3 b3  |&.4b......\{....|

```

Reply:
Hmm, in PNOR there were many zeroes and the first hash(?) is around 0x200 bytes further. Where does the original coreboot image begins? It should be at 0x1200 with ECC